getter.go

```diff
@@ -108,19 +107,26 @@ func newGetter(getURL url.URL, c *Config, b *Bucket) (io.ReadCloser, http.Header
 
 func (g *getter) retryRequest(method, urlStr string, body io.ReadSeeker) (resp *http.Response, err error) {
 	for i := 0; i < g.c.NTry; i++ {
+		time.Sleep(time.Duration(math.Exp2(float64(i))) * 100 * time.Millisecond) // exponential back-off
 		var req *http.Request
 		req, err = http.NewRequest(method, urlStr, body)
 		if err != nil {
-			return
+			logger.debugPrintf("NewRequest error on attempt %d: retrying url: %s, error: %s", i, urlStr, err)
+			continue // let retry
 		}
 		g.b.Sign(req)
 		resp, err = g.c.Client.Do(req)
-		if err == nil {
+
+		// OK status
+		if err == nil && resp != nil && resp.StatusCode == 200 {
```

Comment:
i checked the chunked requests and those us an HTTP client directly, they do not go through this method